### ISSUE TITLE
Device not connected to local push notifications

### ISSUE THREAD

**Issue by kub3let** [home-assistant/android#4363]
**Home Assistant Android app version(s):**
2024.4.1-minimal (from f-droid)

**Android version(s):**
11

**Device model(s):**
surya

**Home Assistant version:**
core 2024.4.3

**Last working Home Assistant release (if known):**

**Description of problem, include YAML if issue is related to notifications:**

I'm making heavy usage of websocket push notifications (not using cloud integration!)

My Homeassistant instance is exposed publicly with my own domain via https and the app is set to `Persistent connection: Always` with only the external domain set, internal domain is left empty.

Every couple days they randomly stop working and if a automation is triggered it logs the following error in HomeAssistant
`2024-04-22 16:20:48.366 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Error for call_service at pos 1: Device not connected to local push notifications`

Just opening the app does not fix it, the app works but no push messages are received, it is not reconnecting automatically.

Killing the android app and opening it up again immediately fixes it but it usually takes me some time to notice the problem. 

Restarting Homeassistant for example also works, in that case it reconnects just fine, I assume it's some weird edge case / deadlock.

**Companion App Logs:**

I just wiped and reinstalled the app in hope that it helps, I will add logs once the problem happens again.

**Screenshot or video of problem:**

**Additional information:**

**Comment by dshokouhi** [comment]
> I just wiped and reinstalled the app in hope that it helps, I will add logs once the problem happens again.

yes please do add the logs they are required for troubleshooting this issue. We will need logs at the time of occurence. 

When this happens does the persistent notification stay or disappears?

**Comment by kub3let** [comment]
> When this happens does the persistent notification stay or disappears?

I actually don't since I had it hidden, I made it visible again to check on that.

Will update in the following days, thanks for the quick reply.

**Comment by nazar-pc** [comment]
I have the same exact issue. The app works, persistent connection is "Always", yet by Home Assistant says "Device not connected to local push notifications". As mentioned above only killing app and starting over fixes the issue until it happens again.

Clearly app either doesn't realize it is disconnected or is not able to reconnect for some reason, which either way looks like a bug to me.

Here is a log from the Android app, hope it is helpful: 
[homeassistant_companion_log_5-8-2024_21-20-47.txt](https://github.com/user-attachments/files/15749279/homeassistant_companion_log_5-8-2024_21-20-47.txt)

**Comment by dshokouhi** [comment]
> Here is a log from the Android app, hope it is helpful:
> [homeassistant_companion_log_5-8-2024_21-20-47.txt](https://github.com/user-attachments/files/15749279/homeassistant_companion_log_5-8-2024_21-20-47.txt)

unfortunately not helpful but it does show the app is connected to the server as all our ping messages to the server are responded to. Maybe this is a HA core issue? Wonder if iOS has a similar issue? 🤔

**Comment by keylogthis** [comment]
This is still a problem and shouldn't have been closed, I have a Pixel 7 Pro and a Pixel 6 and the app does not stay running. Battery optimization is turned off via Allow Background Usage > Unrestricted.

I have other apps that run in the background with the same settings: KDE Connect, Tasker, JamesDSP and those keep running all the time and don't get closed by the battery optimization. This is specific to Home Assistant and not the battery optimization in general on the phones. Home Assistant is the only application that has the problem.

**Comment by nazar-pc** [comment]
Still having this issue here on Pixel 7 Pro more than 1.5 years later. Home Assistant app is running fine, battery optimization is disabled, yet I still get those errors and notification are not working until application restart.

IMO the Home Assistant Core must have a queue and support some basic retries. One-shot notifications are clearly not working properly.

**Comment by MyWebmasterD** [comment]
same issue here with samsung tablet, after a couple of days I get the error Device not connected to local push notifications

**Comment by grimmute** [comment]
Same issue here on Xiaomi 12X, it was working fine until today, now it keeps saying device not connected.
